### PR TITLE
refactor(checkbox): Create BaseCheckbox component

### PR DESCRIPTION
**Pull request by massao**
# Purpose of PR

Create the `BaseCheckbox` component that will handle the `Checkbox`, `RadioButton` and `Switch` components:

image

`BaseCheckbox` uses the same interface as `BaseInput` but since we have the custom checkmarks, we had to create a separate component to be able to handle it properly and keep it accessible.

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/components/inputs/src/base-checkbox/types.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import type { ComponentProps } from 'react';
+import type { BaseInputInternalProps } from '../base-input/types';
+
+export type checkboxTypes = 'checkbox' | 'radio' | 'switch';
+
+export interface BaseCheckboxProps
+  extends Omit<
+    BaseInputInternalProps,
+    'type' | 'as' | 'placeholder' | 'isReadOnly' | 'icon' | 'isInvalid'
```
Comment (suevalov) [inline packages/components/inputs/src/base-checkbox/types.ts:9]: I'm not sure if we need to exclude `isInvalid` here

File: packages/components/inputs/src/base-checkbox/types.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  isChecked?: boolean;
+  /**
+   * Defines if the state is indeterminate
+   * @default false
+   */
+  isIndeterminate?: boolean;
+  /**
+   * Additional props that are passed to the input element
+   */
+  inputProps?: Partial<ComponentProps<'input'>>;
```
Comment (suevalov) [inline packages/components/inputs/src/base-checkbox/types.ts:29]: Do we still need this? If yes, then in what cases?

**Review by suevalov — COMMENTED** [review]
I looks really good in general, but I have one strong IMO that all refs should go directly to input not to label.

File: packages/components/inputs/src/base-checkbox/types.ts (hunk 1 above)
Comment (burakukula) [reply packages/components/inputs/src/base-checkbox/types.ts:9]: yeah, I think we actually gonna need it.

**Review by suevalov — APPROVED** [review]
All comments going to be addressed in the follow up PR

**Review by burakukula — APPROVED** [review]

**Pull request merged**